### PR TITLE
add JSON dump functionality

### PR DESCRIPTION
**Pull request by sam-dixon**

### CODE REVIEW

**Comment by sam-dixon** [comment]
NB: moods + weights aren't yet implemented. `harry.json` has all of the 'harry' memories from book1

File: pensieve/json_dump.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        save: path to save JSON to [default: None]
+
+    Returns:
+        mem_json: JSON object for memory
+    """
+    node = {'name': '',
+            'label': '',
+            'imageURL': mem_dict['img_url'],
+            'iconURL': '',
+            'created': '',
```
Comment (cindeem) [inline pensieve/json_dump.py:27]: This should have a datetime object...for this set we can use utc time
```
from datetime import datetime
created = updated = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.000Z')
```

'2017-07-30T03:03:03.000Z'

File: pensieve/json_dump.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            'updated': ''}
+    relation = {'weight': 0.0,
```
Comment (cindeem) [inline pensieve/json_dump.py]: default weight here should be 0.5

rest at 0.0 is perfect

File: pensieve/json_dump.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        for concept_item in concept_items:
+            clean_text = concept_item.replace(' ', '_')
+            clean_text = clean_text.lower()
+            concept = {'node': {}, 'relation': {}}
+            concept['node'] = {'concept': CONCEPT_MAP[concept_type],
+                               'name': clean_text,
+                               'label': '',
+                               'iconURL': '',
+                               'imageURL': '',
+                               'contact': ''}
```
Comment (cindeem) [inline pensieve/json_dump.py]: contact is no longer used...(deprecated)
you can leave this out, it is not required

File: pensieve/json_dump.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                                   'originType': 'OriginUserDefined',
+                                   'joy': 0.0,
+                                   'fear': 0.0,
+                                   'surprise': 0.0,
+                                   'sadness': 0.0,
+                                   'disgust': 0.0,
+                                   'anger': 0.0}
+            concepts.append(concept)
+    narrative = {'node': {'name': '',
+                          'label': '',
```
Comment (cindeem) [inline pensieve/json_dump.py]: lets default 'label': 'title'
In the game this becomes the default text so user knows they can update

File: pensieve/json_dump.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                          'text': mem_dict['narrative']},
+                 'relation': {'weight': 0.5}}
+    mem = {'memory': '',
+           'node': node,
+           'relation': relation,
+           'concepts': concepts,
+           'narrative': narrative}
+
+    if save is not None:
```
Comment (cindeem) [inline pensieve/json_dump.py:77]: We can also save one file as a list of mem dictionaries...but no need to change I can work with either

File: pensieve/pensieve.py [hunk 6]
```diff
@@ -180,10 +195,10 @@ def find_character_paragraphs(self, char_name, density_cut=0.8):
                 char_pars.append(par)
         return char_pars
 
-    def build_doc_memories(self, char_name, density_cut=0.8,
-                           verb_cut=500, name_cut=100):
+    def gather_doc_memories(self, char_name, density_cut=0.8,
+                            verb_cut=500, name_cut=100, save=None):
         """
-        Builds memories from character paragraphs.
+        Collects memories from character paragraphs.
```
Comment (cindeem) [inline pensieve/pensieve.py:201]: 👍

File: pensieve/pensieve.py [hunk 7]
```diff
@@ -353,18 +380,20 @@ def search_for_images(self):
...
-
-    from pprint import pprint
-    print('{} paragraphs found'.format(len(harry_pars)))
-    rand_index = randint(0, len(harry_pars)-1)
-    test_par = harry_pars[rand_index]
-    print('Book '+str(test_par.doc.id+1))
-    print('Paragraph '+str(test_par.id))
-    print(test_par.text)
-    pprint(test_par.words)
+    # try:
```
Comment (cindeem) [inline pensieve/pensieve.py:383]: Should consider getting rid of the commented out code....but again not necessary

**Review by cindeem — COMMENTED** [review]
Looks really good...
I have a couple suggested changes...(small updates)..then it should be good to merge
good job!

**Pull request merged**